core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java

```diff
@@ -2949,6 +2956,9 @@ private String getTableName(String alias, RelNode rel) {
     } else {
       tableName = alias;
     }
+    if (tableName == null && alias != null) {
```

Comment:
Why is this check required if in else part we are always setting - tableName = alias;